crates/driver/src/infra/notify/mod.rs

```diff
@@ -24,23 +24,21 @@ pub fn scoring_failed(
     }
 
     let notification = match err {
-        score::Error::ObjectiveValueNonPositive => {
-            notification::Kind::ScoringFailed(notification::ScoreKind::ObjectiveValueNonPositive)
-        }
         score::Error::ZeroScore => {
             notification::Kind::ScoringFailed(notification::ScoreKind::ZeroScore)
         }
-        score::Error::ScoreHigherThanObjective(score, objective_value) => {
-            notification::Kind::ScoringFailed(notification::ScoreKind::ScoreHigherThanObjective(
-                *score,
-                *objective_value,
-            ))
-        }
-        score::Error::SuccessProbabilityOutOfRange(success_probability) => {
-            notification::Kind::ScoringFailed(
-                notification::ScoreKind::SuccessProbabilityOutOfRange(*success_probability),
-            )
+        score::Error::ScoreHigherThanQuality(score, quality) => notification::Kind::ScoringFailed(
+            notification::ScoreKind::ScoreHigherThanQuality(*score, *quality),
+        ),
+        score::Error::RiskAdjusted(score::risk::Error::SuccessProbabilityOutOfRange(
+            success_probability,
+        )) => notification::Kind::ScoringFailed(
+            notification::ScoreKind::SuccessProbabilityOutOfRange(*success_probability),
+        ),
+        score::Error::RiskAdjusted(score::risk::Error::ObjectiveValueNonPositive) => {
+            notification::Kind::ScoringFailed(notification::ScoreKind::ObjectiveValueNonPositive)
         }
+        score::Error::RiskAdjusted(score::risk::Error::Boundary(_)) => return,
         score::Error::Boundary(_) => return,
```

Comment:
Is it expected that we don't notify about anything if any of these errors occur?

Reply:
For now yes, it is expected. We still need to figure out the boundary line - which kind of internal errors we want to propagate to solvers.